compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4966,41 +4966,57 @@ object Types {
   }
 
   class TypeSizeAccumulator(implicit ctx: Context) extends TypeAccumulator[Int] {
-    def apply(n: Int, tp: Type): Int = tp match {
-      case tp: AppliedType =>
-        foldOver(n + 1, tp)
-      case tp: RefinedType =>
-        foldOver(n + 1, tp)
-      case tp: TypeRef if tp.info.isTypeAlias =>
-        apply(n, tp.superType)
-      case tp: TypeParamRef =>
-        apply(n, ctx.typeComparer.bounds(tp))
-      case _ =>
-        foldOver(n, tp)
+    val seen: util.HashSet[Type] = new util.HashSet[Type](64) {
```

Comment:
It's probably faster to use a `java.util.IdentityHashMap[Type, Type]`, We already use that in lots of other places in the compiler.

Reply:
Or even:
```scala
import java.util.{Collections, IdentityHashMap}
val seen = Collections.newSetFromMap[Type](new IdentityHashMap)
```